nestkernel/simulation_manager.cpp

```diff
@@ -692,10 +736,25 @@ nest::SimulationManager::update_connection_infrastructure( const thread tid )
     }
   }
 
+#ifdef TIMER_DETAILED
+  if ( tid == 0 )
+  {
+    sw_gather_target_data_.start();
+  }
+#endif
+
   // communicate connection information from postsynaptic to
   // presynaptic side
   kernel().event_delivery_manager.gather_target_data( tid );
 
+#ifdef TIMER_DETAILED
+#pragma omp barrier
```

Comment:
Same as above.

Reply:
Yes, it's not frequently called. Keep this barrier.